src/App.tsx

```diff
@@ -21,12 +21,12 @@ const App = observer(() => {
             <div key={task.id}>
               <input type="checkbox"
                 checked={task.completed}
-                onChange={action(e => task.completed = e.target.checked)} />
+                onChange={e => task.completed = e.target.checked} />
               <input
                 value={task.title}
-                onChange={action(e => task.title = e.target.value)} />
+                onChange={e => task.title = e.target.value} />
               <button type="button" onClick={() => store.saveTask(task)}>Save</button>
```

Comment:
Shouldn't this also become `task.save()`?

Reply:
I was wondering about that.

In the current implementation, when the save fails - it displays an alert.
Initially, it was placed in the Store.
```ts
  async saveTask(task: Task) {
    try {
      await task.save();
    } catch (error: any) {
      alert(error.message);
    }
  }
```


I moved it to the button handler:

```ts
              <button type="button" onClick={() => {
                try {
                  store.saveTask(task)
                } catch (error: any) {
                  alert(error.message);
                }
              }}>Save</button>

```
and then it looked like too much code in the handler and returned it to the store.

What do you think?

Reply:
I think that that's how the sausage is made. This sort of code has to live somewhere. What I do is:
- Keep biz logic outside of the UI. The only thing this component should do is accept a prop of type `VoidFunction` or accepting whatever data oriented argument(s) it needs.
- Keep the runtime platform outside of the model/store: so no `MouseEvent` or `EventTarge` or anything DOM/Node/RN/whatever platform can exist in biz entities. Only pure JS/TS types and dependencies.
- Given that, where does the `alert(error.message)` go? For one-off use like here I'd put it in the same component file and wrap with `reaction` - just cause it isn't a react component. Normally nobody uses `alert` though, which makes it a bit contrived and the "normal" UI is a react component or a toast package. If you need to call a platform dependent API like a toast or similar, then I usually do this:
```ts
class SomeModel {
  maybeToastProvider: Toast | null = null
  setToastProvider = (toastProvider: Toast): void => {
    this.toastProvider = toastProvider
  }
```
And then as part of the instantiation of `SomeModel`, I pass the `Toast` thing into it: `SomeModel.create({ whatever, the, domain, data, andAlso, toastProvider })`. Then in `SomeModel`'s methods you call `this.toastProvider?.info(message)`.